### PR TITLE
Feature/add twilio support

### PR DESCRIPTION
**Pull request by gauravtiwari**

### CODE REVIEW

File: lib/cellular/backends.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ module Backends
     autoload :Backend, 'cellular/backends/backend'
     autoload :CoolSMS, 'cellular/backends/cool_sms'
     autoload :Sendega, 'cellular/backends/sendega'
+    autoload :Twilio, 'cellular/backends/twilio'
```
Comment (toothfairy) [inline lib/cellular/backends.rb]: indentation is broken here - it's a very important high priority thing!

**Comment by toothfairy** [comment]
Looks pretty good to me

File: lib/cellular/backends.rb (hunk 1 above)
Comment (gauravtiwari) [inline lib/cellular/backends.rb]: Oops! Fixed 🎉

**Comment by jgorset** [comment]
Yeah, really good!

**Comment by timkurvers** [comment]
Is Guard no longer compatible with Ruby or why is it being removed? If we are indeed dropping, make sure to get rid of the `Guardfile`, too.

**Comment by gauravtiwari** [comment]
@timkurvers It's using Listen, which unfortunately is only supporting ruby 2.2+. Cellular supports 2.0+

**Comment by timkurvers** [comment]
Strikes me as odd that those two gems wouldn't adhere to semantic versioning 😮

**Comment by gauravtiwari** [comment]
@timkurvers Yeah, more like Rails 5, which is now only supports ruby 2.2+ https://rubygems.org/gems/listen/versions/3.1.1

**Comment by gauravtiwari** [comment]
Probably, because of Rails5, as it uses listen in development for file watching now (sort of optional dependency, but it's there in Gemfile)

**Comment by timkurvers** [comment]
They seem to adhere to semantic versioning, but Guard has a very loose dependency constraint: https://github.com/guard/guard/blob/master/guard.gemspec#L20

Locking Listen to 2.x would probably work just fine, unless other dependencies require a newer version.

**Comment by gauravtiwari** [comment]
Yup, I kinda of realised while discussing this with you :), but not sure if we really need guard?

**Comment by timkurvers** [comment]
I haven't done any Ruby development in a while, but Guard was the de-facto standard for ensuring tests run automatically. It depends on what people are using to run their tests nowadays, other than manually. 😄

**Comment by timkurvers** [comment]
Also just realized we haven't really been following the Better Specs guidelines in this repository with regards to [usage of should](http://betterspecs.org/#should) so I propose we leave that for another time.

**Comment by gauravtiwari** [comment]
haha yeah - automated red-green thing, but I guess it would be good to be keep the Gem lean without many deps.

File: spec/cellular/backends/twilio_spec.rb [hunk 2]
```diff
@@ -44,15 +44,15 @@
        to_return(:status => [201, "CREATED"], :body => fixture('backends/twilio/success.json'), :headers => {'Content-Type' => 'application/json'})
     end
 
-    it 'uses HTTParty to deliver an SMS' do
+    it 'does uses HTTParty to deliver an SMS' do
```
Comment (timkurvers) [inline spec/cellular/backends/twilio_spec.rb]: This should be `it 'uses ...'`.

File: spec/cellular/backends/twilio_spec.rb [hunk 3]
```diff
@@ -44,15 +44,15 @@
...
       expect(HTTParty).to receive(:post).with(described_class::sms_url, body:
         payload, headers: described_class::HTTP_HEADERS, basic_auth: described_class::twilio_config).and_call_original
 
       described_class.deliver(options)
     end
 
     context 'when successful' do
-      it 'returns status code and a message' do
+      it 'does return a status code and message' do
```
Comment (timkurvers) [inline spec/cellular/backends/twilio_spec.rb]: This should be `it 'returns ...'`.

**Comment by timkurvers** [comment]
Agree, but it's a development dependency.

**Comment by gauravtiwari** [comment]
Yeah, Should we add it back then? Travis failed a couple of times today and I spent sometime figuring out what's wrong, that's why removed it. Sure, we could lock down versions, but in future, it might change again, and we have to scratch our heads so would be good to move away to a few deps if possible, even development Gems. 

Rspec, manually runs quite quick OR perhaps another approach would be to have a Guardfile_example file in repo, which could be used incase folks want to use Guard? What you think?

**Comment by timkurvers** [comment]
I'd probably roll with removal, given that having a dirty working directory is not something people would like to have. 👍

**Pull request merged**